### PR TITLE
chore: upgrading to angular v10

### PR DESCRIPTION
**Pull request by bogusweb**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/spectator/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Currenty spectator is on Angular 9. Its need to be updated since Angular start using tslib@2.0.0 and cannot update project to v10.
Issue Number: #325 

## What is the new behavior?
Upgraded to Angular 10.

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

- Upgraded angular to v10
- Removed deprecated `Testbed.get`

BREAKING CHANGE: This PR sets the angular peer dependencies to v10

### CODE REVIEW

**Comment by bogusweb** [comment]
Draft because i have to finish:

- [ ] Update docs
- [ ] `npm run build` dont work, error below
- [ ] Change `createService` to `createServiceFactory`, but tslint cries `property 'inject' does not exist on type 'SpectatorServiceFactory '`. 


#### Build error
Maybe someone know this bug, for now i cant fix this 😭 

```------------------------------------------------------------------------------
Building entry point '@ngneat/spectator/jest'
------------------------------------------------------------------------------
Compiling TypeScript sources through ngc
ERROR: dist/spectator/public_api.d.ts:1:23 - error TS2688: Cannot find type definition file for 'projects/spectator/src/lib/matchers-types'.

1 /// <reference types="projects/spectator/src/lib/matchers-types" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

An unhandled exception occurred: dist/spectator/public_api.d.ts:1:23 - error TS2688: Cannot find type definition file for 'projects/spectator/src/lib/matchers-types'.

1 /// <reference types="projects/spectator/src/lib/matchers-types" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by adam-beck** [comment]
Is the build error caused by https://github.com/ng-packagr/ng-packagr/issues/1654

**Comment by bogusweb** [comment]
thanks for the information.
I wonder if after downgrade these two packages will build a library sufficiently for the current and previous versions angular 🤔

**Comment by bogusweb** [comment]
there are changes in schematics folder that I don't know if they are needed. These changes are in a separate commit

**Comment by NetanelBasal** [comment]
Can you summarize the changes and why you did each one of them, please?

**Comment by bogusweb** [comment]
- Upgrade to Angular 10 and dependencie tslib to 2.0.0, #325 
- Removed deprecated `TestBed.get` for `TestBed.inject`. But I left support for lower versions that don't have `TestBed.inject`
- Removed announced removal `createService` and update tests with `createServiceFactory`
- Updated all tests where it was used `spectator.get` for `spectator.inject` (also for jest)
- Updated all docs
- Upgraded ng-packagr to latest verrsion and change custom referenced typings from `*.d.ts` to `*.ts`, [here information why](https://github.com/ng-packagr/ng-packagr/issues/1727#issuecomment-678085045)

and one last thing I'm not sure is changes in schematics after `build:schematics` command.

Everything tested, additionally checked in a private project if all types work and if the library works.

**Comment by NetanelBasal** [comment]
I think this is an opportunity to move to version 5. We can support only v8+ and remove the deprecated APIs. Do you think you can do it?

**Comment by bogusweb** [comment]
u mean version 6? Because already spectator is in version 5 :)
If so, in the evening I will remove deprecated APIs and fix tests.

**Comment by NetanelBasal** [comment]
Yes, v6. The comments in the code confused me :)

Yes, but pay attention that when you run `npm run commit`, you should choose the breaking change option, and write the changes.

Thanks!

**Comment by bogusweb** [comment]
angular peer dependencies already has `>= 8.0.0`? So it doesn't have to be changed, does it?

**Comment by NetanelBasal** [comment]
Yeap

**Comment by bogusweb** [comment]
I think I've finished, there are no more things that were marked as deprecated.

Unfortunately this is my first pull request to open source library so I would ask you to check if it is well done.

File: projects/spectator/src/lib/spectator-routing/create-factory.ts [hunk 1]
```diff
@@ -86,5 +88,18 @@ function createSpectatorRouting<C>(options: Required<SpectatorRoutingOptions<C>>
 
   const component = setProps(fixture.componentInstance, props);
 
-  return new SpectatorRouting(fixture, debugElement, component, TestBed.get(Router), TestBed.get(ActivatedRoute));
+  /**
+   * Back compatibility, angular under 8 version doesnt have a inject function
+   */
+  if (!TestBed.inject) {
```
Comment (NetanelBasal) [inline projects/spectator/src/lib/spectator-routing/create-factory.ts:94]: ?

Comment (bogusweb) [reply projects/spectator/src/lib/spectator-routing/create-factory.ts:94]: Mistake in comment, `inject` in TestBed was added in Angular 9 version so we need this backward compatibility. I'm going to correct the comments

Comment (bogusweb) [reply projects/spectator/src/lib/spectator-routing/create-factory.ts:94]: updated comments

File: tslint.json [hunk 2]
```diff
@@ -56,8 +56,6 @@
     "no-unbound-method": false,
     "no-unnecessary-callback-wrapper": false,
     "no-unsafe-any": false,
-    "no-unused-variable": false,
```
Comment (NetanelBasal) [inline tslint.json:59]: Why removing those?

File: projects/spectator/tsconfig.spec.json [hunk 3]
```diff
@@ -22,6 +22,6 @@
   ],
   "include": [
     "test/**/*.spec.ts",
-    "src/lib/matchers-types.d.ts"
+    "src/lib/matchers-types.ts"
```
Comment (NetanelBasal) [inline projects/spectator/tsconfig.spec.json:25]: Why?

**Comment by NetanelBasal** [comment]
LGTM. 

@benelliott It will be helpful if you can go over it too.

File: tslint.json (hunk 2 above)
Comment (bogusweb) [reply tslint.json:59]: cli this remove after update, this rule is deprecated: https://github.com/palantir/tslint/pull/3919

File: projects/spectator/tsconfig.spec.json (hunk 3 above)
Comment (bogusweb) [reply projects/spectator/tsconfig.spec.json:25]: https://github.com/ng-packagr/ng-packagr/issues/1727#issuecomment-678085045

**Comment by NetanelBasal** [comment]
@bogusweb thanks for the work. Did you test it on a real project? Including the custom matchers types.

**Comment by bogusweb** [comment]
yeah, by symlink and by replacing spectator in node-modules :D Types from matcher-types working and tests passed

**Comment by NetanelBasal** [comment]
Cool, thanks. I'll wait one more day for someone else who wants to review and merged it. Don't forget to add your name to the contributors' list.

**Comment by benelliott** [comment]
I haven't had a chance to run it on a project, but the diff looks good to me. Thanks @bogusweb !

**Comment by NetanelBasal** [comment]
Can you please list the breaking changes for this version [here](https://github.com/ngneat/spectator/blob/master/BREAKING_CHANGES.md)?
